mmcv/runner/hooks/optimizer.py

```diff
@@ -66,37 +89,56 @@ def __init__(self,
         self.coalesce = coalesce
         self.bucket_size_mb = bucket_size_mb
         self.distributed = distributed
+        self._use_torch_amp = (
+            TORCH_VERSION != 'parrots' and TORCH_VERSION >= '1.6.0')
+
         if loss_scale == 'dynamic':
-            self.loss_scaler = LossScaler(mode='dynamic')
+            self._scale_update_param = None
+            if self._use_torch_amp:
+                self.loss_scaler = GradScaler(init_scale=loss_scale)
+            else:
+                self.loss_scaler = LossScaler(mode='dynamic')
         elif isinstance(loss_scale, float):
-            self.loss_scaler = LossScaler(init_scale=loss_scale, mode='static')
+            self._scale_update_param = loss_scale
+            if self._use_torch_amp:
+                self.loss_scaler = GradScaler(init_scale=loss_scale)
+            else:
+                self.loss_scaler = LossScaler(
+                    init_scale=loss_scale, mode='static')
         elif isinstance(loss_scale, dict):
-            self.loss_scaler = LossScaler(**loss_scale)
+            self._scale_update_param = None
```

Comment:
Can we simplify the relationship between `self._scale_update_param` and `self.loss_scaler`?

Reply:
Good idea. I have another option in mind, will have a try.